wgengine/router/dns/nm.go

```diff
@@ -52,15 +53,82 @@ func isNMActive() bool {
 
 // nmManager uses the NetworkManager DBus API.
 type nmManager struct {
+	logf          logger.Logf
 	interfaceName string
 }
 
 func newNMManager(mconfig ManagerConfig) managerImpl {
 	return nmManager{
+		logf:          mconfig.Logf,
 		interfaceName: mconfig.InterfaceName,
 	}
 }
 
+const nmDeviceActiveState = 100
```

Comment:
document what this means. I assume it's a magic status number from network manager, but I'm not certain.